pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -114,3 +121,24 @@ func (r *ReconcileNMstate) syncKubernetesNMStateCRDs(instance *nmstatev1alpha1.N
 	}
 	return nil
 }
+
+func (r *ReconcileNMstate) addControllers(instance *nmstatev1alpha1.NMstate) error {
+	var err error
+
+	cfg, err := config.GetConfig()
```

Comment:
We don't need to use client-go directly here AFAIK. operator-sdk gives us controller-runtime client which is supposed to be used in this framework. https://github.com/operator-framework/operator-sdk/blob/11aa2af7aaa202cb38b2417031e6af2035c7948c/doc/user/client.md

Reply:
AFAICT that's what's happening here... Look at the imports. Am I missing something?

Reply:
I'm not sure I understand what you plan to do in this structure. We cannot keep nmstate controllers (the existing ones) running as part of the operator process. The operator should merely deploy manifests. So here you would just render templates for our existing daemon sets and apply them.

The operator should be decoupled from the rest of the processes by this "create manifest" interface. So we can later simply split operator from the rest if needed.

Reply:
I think we are good, the PR also checks RUN_OPERATOR add the non operator controllers so they are not run, so we can run the kubernetes-nmstate-handler in two modes, running only the operator controller and running only the non-operator controllers.

That's right @bcrochet

Reply:
But with this code, we will run nmstate controllers as a part of the operator, that is wrong. The operator should just apply manifests with operands.

Reply:
Ahh wait, that's right, why do we do this ?